core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -19,122 +19,116 @@
 import org.apache.iceberg.exceptions.RuntimeIOException;
 
 /**
- * The HadoopCatalog provides a way to manage the path-based table just like catalog. It uses
- * a specified directory under a specified filesystem as the warehouse directory, and organizes
+ * HadoopCatalog provides a way to use table names like db.table to work with path-based tables under a common
+ * location. It uses a specified directory under a specified filesystem as the warehouse directory, and organizes
  * two levels directories that mapped to the database and the table respectively. The HadoopCatalog
- * takes a URI as its root directory, and creates a child directory "iceberg/warehouse" as the
- * warehouse directory. When creating a table such as $db.$tbl, it creates $db.db/$tbl under
+ * takes a URI as the warehouse directory. When creating a table such as $db.$tbl, it creates $db/$tbl under
  * iceberg/warehouse, and put the table metadata into the directory.
  *
- * The HadoopCatalog now supports createtable, droptable, the renametable is not supported yet.
+ * The HadoopCatalog now supports {@link org.apache.iceberg.catalog.Catalog#createTable},
+ * {@link org.apache.iceberg.catalog.Catalog#dropTable}, the {@link org.apache.iceberg.catalog.Catalog#renameTable}
+ * is not supported yet.
  *
- * Note: The HadoopCatalog requires the rename operation of the filesystem is an atomic operation.
+ * Note: The HadoopCatalog requires that the underlying file system supports atomic rename.
  */
 public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
   private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
   private final Configuration conf;
-  private String warehouseUri;
+  private String warehouseLocation;
+
+  public HadoopCatalog(Configuration conf, String warehouseLocation) {
+    Preconditions.checkArgument(warehouseLocation != null && !warehouseLocation.equals(""),
+        "no location provided for warehouse");
 
-  public HadoopCatalog(Configuration conf, String warehouseUri) {
     this.conf = conf;
+    this.warehouseLocation = warehouseLocation.replaceAll("/$", "");
+  }
 
-    if (warehouseUri != null) {
-      this.warehouseUri = warehouseUri;
-    } else {
-      String fsRoot = conf.get("fs.defaultFS");
-      Path warehousePath = new Path(fsRoot, ICEBERG_HADOOP_WAREHOUSE_BASE);
-      try {
-        FileSystem fs = Util.getFs(warehousePath, conf);
-        if (!fs.isDirectory(warehousePath)) {
-          if (!fs.mkdirs(warehousePath)) {
-            throw new IOException("failed to create warehouse for hadoop catalog");
-          }
+  public HadoopCatalog(Configuration conf) {
+    String fsRoot = conf.get("fs.defaultFS");
+    Path warehousePath = new Path(fsRoot, ICEBERG_HADOOP_WAREHOUSE_BASE);
+
+    try {
+      FileSystem fs = Util.getFs(warehousePath, conf);
+      if (!fs.isDirectory(warehousePath)) {
+        if (!fs.mkdirs(warehousePath)) {
```

Comment:
Why does this need to create the warehouse path? I don't think this is necessary.